File: tests/property/python/areas/polynomials.py
"""SageMath side of the ``polynomials`` property-test area (GF(p)[x]).

Cases: tests/property/cases/polynomials.cases.json
"""

from sage.all import *

from ._helpers import (
    poly_add,
    poly_derivative,
    poly_eval,
    poly_factor,
    poly_gcd,
    poly_is_irreducible,
    poly_mod,
    poly_mul,
    poly_pow,
    poly_quo_rem,
    poly_roots,
)

FUNCTIONS = {
    'poly_add': poly_add,
    'poly_mul': poly_mul,
    'poly_quo_rem': poly_quo_rem,
    'poly_mod': poly_mod,
    'poly_gcd': poly_gcd,
    'poly_eval': poly_eval,
    'poly_factor': poly_factor,
    'poly_derivative': poly_derivative,
    'poly_is_irreducible': poly_is_irreducible,
    'poly_roots': poly_roots,
    'poly_pow': poly_pow,
}

File: tests/property/typescript/areas/polynomials.ts
/**
 * sagemath-ts side of the `polynomials` property-test area (GF(p)[x]).
 *
 * Cases: tests/property/cases/polynomials.cases.json
 * SageMath counterpart: tests/property/python/areas/polynomials.py
 */

import {
  polyAdd,
  polyDerivative,
  polyEval,
  polyFactor,
  polyGcd,
  polyIsIrreducible,
  polyMod,
  polyMul,
  polyPow,
  polyQuoRem,
  polyRoots,
} from './_helpers.js';

export const functions = {
  poly_add: (p: bigint, coeffs1: bigint[], coeffs2: bigint[]) => polyAdd(p, coeffs1, coeffs2),
  poly_mul: (p: bigint, coeffs1: bigint[], coeffs2: bigint[]) => polyMul(p, coeffs1, coeffs2),
  poly_quo_rem: (p: bigint, coeffs1: bigint[], coeffs2: bigint[]) =>
    polyQuoRem(p, coeffs1, coeffs2),
  poly_mod: (p: bigint, coeffs1: bigint[], coeffs2: bigint[]) => polyMod(p, coeffs1, coeffs2),
  poly_gcd: (p: bigint, coeffs1: bigint[], coeffs2: bigint[]) => polyGcd(p, coeffs1, coeffs2),
  poly_eval: (p: bigint, coeffs: bigint[], x: bigint) => polyEval(p, coeffs, x),
  poly_factor: (p: bigint, coeffs: bigint[]) => polyFactor(p, coeffs),
  poly_derivative: (p: bigint, coeffs: bigint[]) => polyDerivative(p, coeffs),
  poly_is_irreducible: (p: bigint, coeffs: bigint[]) => polyIsIrreducible(p, coeffs),
  poly_roots: (p: bigint, coeffs: bigint[]) => polyRoots(p, coeffs),
  poly_pow: (p: bigint, coeffs: bigint[], n: bigint) => polyPow(p, coeffs, n),
};
